sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/MethodOverloadsShouldBeGrouped.cs

```diff
@@ -274,4 +279,68 @@ protected override void DoSomething(bool b)
 
     }
 
+    public class InterfaceImplementationTogether : IEqualityComparer<int>, IEqualityComparer<string>
+    {
+        public bool Equals(int x, int y) { return true; }
+
+        public int GetHashCode(int obj) { return 0; }
+
+        public bool Equals(string x, string y) { return true; }
+
+        public int GetHashCode(string obj) { return 0; }
+    }
+
+    public interface ITest<TItem>
+    {
+        void Aaa(TItem x);
+        event EventHandler<TItem> Eee;
+        TItem this[TItem index] { get; set; }
+        TItem Ppp { get; set; }
+        void Zzz(TItem x);
+    }
+
+    public class InterfaceImplementationTogetherWithPropertiesAndEvents : ITest<int>, ITest<string>
+    {
+        public event EventHandler<int> Eee;
+        public void Aaa(int x) { }
+        public int this[int index] { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+        int ITest<int>.Ppp { get; set; }
+        public void Zzz(int x) { }
+
+        event EventHandler<string> ITest<string>.Eee
+        {
+            add { }
+            remove { }
+        }
+        public void Aaa(string x) { }
+        public string this[string index] { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+        public string Ppp { get; set; }
+        public void Zzz(string x) { }
+    }
+
+    public class ImplicitInterfaceWithEventHandler : ITest<int>
+    {
+        public void Zzz(string x) { } // Noncompliant
+
+        // Implicit interface implementation groupped together
+        public void Aaa(int x) { }
+        int ITest<int>.Ppp { get; set; }
+        public int this[int index] { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+        public event EventHandler<int> Eee; // C# events are not recognized as parts of interface group, grouping is splitted here
+        public void Zzz(int x) { } // Secondary
+    }
+
+    public interface ICancel
+    {
+        void Cancel();
+        void Cancel(bool b);
+        void Renew();
+    }
+
+    public class InterfaceImplementationMethodsTogether : ICancel
```

Comment:
I've noticed that the ctor is ignored if it separates the overloads:
```
    public class InterfaceImplementationMethodsTogether : ICancel
    {
        public void Cancel() { } // Noncompliant, it should be adjecent inside same interface implementation group

        public InterfaceImplementationMethodsTogether() { }

        public void Cancel(bool b) { } // Secondary

        public void Renew() { }
    }
```
Isn't this a false negative?

Reply:
Looks fine to me :) Issues are reported.